### PR TITLE
fix: RegEx should be time limited

### PR DESCRIPTION
**Pull request by ynsehoornenborg**
For security reasons.

### CODE REVIEW

**Review by bcollamore — APPROVED** [review]

File: Philips.CodeAnalysis.SecurityAnalyzers/RegexNeedsTimeoutAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+			Title, MessageFormat, Category, 
+			DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterSyntaxNodeAction(Analyze, SyntaxKind.ObjectCreationExpression);
```
Comment (bcollamore) [inline Philips.CodeAnalysis.SecurityAnalyzers/RegexNeedsTimeoutAnalyzer.cs:32]: You can try using the new pattern if desired. Or, save for next time.

File: Philips.CodeAnalysis.SecurityAnalyzers/RegexNeedsTimeoutAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		}
+
+		private void Analyze(SyntaxNodeAnalysisContext context)
+		{
+			var creation = (ObjectCreationExpressionSyntax)context.Node;
+
+
+			// Bail out early.
+			TypeSyntax typeSyntax = creation.Type;
+			if(!typeSyntax.ToString().Contains("Regex"))
```
Comment (bcollamore) [inline Philips.CodeAnalysis.SecurityAnalyzers/RegexNeedsTimeoutAnalyzer.cs:42]: This isn't a magic string?

**Review by bcollamore — APPROVED** [review]

File: Philips.CodeAnalysis.SecurityAnalyzers/RegexNeedsTimeoutAnalyzer.cs (hunk 1 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.SecurityAnalyzers/RegexNeedsTimeoutAnalyzer.cs:32]: Leave this for a next time

File: Philips.CodeAnalysis.SecurityAnalyzers/RegexNeedsTimeoutAnalyzer.cs (hunk 2 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.SecurityAnalyzers/RegexNeedsTimeoutAnalyzer.cs:42]: Nope, only used one time

**Pull request merged**